acct/check

```diff
@@ -132,18 +137,69 @@ if [ -f "$vhost_file" ] && vhost=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$v
 fi
 
 # check apphosting config
-if [ -f "$vhost_app_file" ] && vhost=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_app_file"); then
+if [ -f "$vhost_app_file" ] && vhost_app=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_app_file"); then
   echo
   echo "Virtual apphosts:"
   echo "-------------------------------------------------------------------------------"
-  echo "$vhost"
+  echo "$vhost_app"
+  echo "-------------------------------------------------------------------------------"
+fi
+
+# check virtual mail config
+if [ -f "$vhost_mail_file" ] && vhost_mail=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_mail_file"); then
+  echo
+  echo "Virtual mail:"
+  echo "-------------------------------------------------------------------------------"
+  echo "$vhost_mail"
+  echo "-------------------------------------------------------------------------------"
+fi
+
+
+# check dns entries
+expand_fqdns()
+{
+  while read -r entry; do
+    if [ -z "$entry" ]; then
+      continue
+    fi
+    
+    username=$(echo "$entry" | cut -d' ' -f1)
+    domain=$(echo "$entry" | cut -d' ' -f2)
+
+    # shellcheck disable=SC2143,SC2086,SC2063
+    if [ "$domain" = "-" ]; then
+      echo "$username.berkeley.edu"
+    elif [ "$(echo $domain | grep -E '*\.[a-z]+$')" ]; then
+      echo "$domain"
+    else
+      echo "$domain.berkeley.edu"
+    fi
+  done
+}
+
+# shellcheck disable=SC2028
+fqdns=$(echo "$vhost\n$vhost_app\n$vhost_mail" | grep -v '#' | expand_fqdns)
```

Comment:
this seems ugly to me (and shellcheck doesn't like it) but all the alternatives I tried don't seem to work or fail on certain cases.

Reply:
Have you tried using `printf "%s\n%s\n%s\n" "$vhost" "$vhost_app" "$vhost_mail"`?

btw the way you have it now wouldn't work on bash...